source/dofs/dof_handler.cc

```diff
@@ -3226,27 +3280,21 @@ DoFHandler<dim, spacedim>::post_refinement_action()
 
 template <int dim, int spacedim>
 void
-DoFHandler<dim, spacedim>::pre_active_fe_index_transfer()
+DoFHandler<dim, spacedim>::pre_transfer_action()
 {
-  // Finite elements need to be assigned to each cell by calling
-  // distribute_dofs() first to make this functionality available.
-  if (this->fe_collection.size() > 0)
-    {
```

Comment:
Why can this if-statement go?

Reply:
Initially, fe indices should only be transferred when a corresponding `FECollection` has been registered.

As we now initialize the fe index tables right after construction and allow users to set fe indices to their liking, they need to be transferred without a registered collection. We check if the fe indices are valid when calling `distribute_dofs`.